### PR TITLE
[ADD] PLM: approvals doc

### PR DESCRIPTION
**Pull request by Felicious**
### Summary

Create a new PLM doc detailing:
1. How to add approvers
2. A workflow of accepting/rejecting ECOs
3. Odoo's automatic planned activity creation for approvers

Note for reviewers: there are a few custom anchors that can't be found, and they are to links in the [ECO doc](https://github.com/odoo/documentation/pull/5238) . This PR should be pushed AFTER the ECO PR is pushed, so the links will work accordingly.

### CODE REVIEW

**Comment by Felicious** [comment]
Hello @odoo/manufacturing-doc-review! This is the last PLM doc for Tim's whitepaper :smile: Yayy we're done adding PLM docs! Thank you for the peer review :smiley:

File: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+Approve ECOs
+------------
+
+Navigate to an |ECO| in a verification stage while logged in as the assigned approval to see the
+:guilabel:`Approve`, :guilabel:`Reject`, and :guilabel:`Apply Changes` button. To approve the |ECO|
+and apply the changes onto the production :abbr:`BoM (Bill of Materials)`, click :guilabel:`Approve`
+then :guilabel:`Apply Changes`. Note that the :guilabel:`Apply Changes` button will not work unless
+the :guilabel:`Approve` button was pressed first. Additionally, the chatter will log the history of
+the buttons pressed.
```
Comment (odoo-shke) [inline content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst:85]: apply changets button will not work until all approvers have approvede. (When there are multiple approvers)

**Review by odoo-shke — COMMENTED** [review]
Looks good !

**Comment by StraubCreative** [comment]
@odoo/manufacturing-doc-review review this asap please :)

File: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst (hunk 1 above)
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst:85]:
```suggestion
the :guilabel:`Approve` button was clicked first. Additionally, the chatter will log the history of
the buttons clicked.
```

File: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+.. warning::
+   When the :guilabel:`Approval Type` is **not** set to :guilabel:`Is required to approve`, approval
+   from the associated user is not needed before applying changes with the the :guilabel:`Apply
+   Changes` button. Thus, the :guilabel:`Apply Changes` **will work** without the requiring the
+   :guilabel:`Approve` button to be pressed first. pressed before it.
```
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst:91]:
```suggestion
   from the associated user is not needed before applying changes with the :guilabel:`Apply
   Changes` button. Thus, the :guilabel:`Apply Changes` button **will work** without requiring the
   :guilabel:`Approve` button to be clicked first.
```

File: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+
+Automated activities
+--------------------
+
+When an |ECO| is moved to a verification stage, a planned activity is automatically created for
+assigned approvers to review the |ECO|. Approvers receive a notification in their activities inbox,
+accessible through the :guilabel:`🕘 (clock)` icon at the top of the page. In to-do task list, the
+:guilabel:`Engineering Change Order (ECO)` notification displays the amount of :guilabel:`Late`,
+:guilabel:`Today`'s, and :guilabel:`Future` activities. Clicking on each of these buttons shows a
```
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst:101]:
```suggestion
accessible through the :guilabel:`🕘 (clock)` icon at the top of the page. In the to-do task list, the
:guilabel:`Engineering Change Order (ECO)` notification displays the number of 
activities marked :guilabel:`Late`,
:guilabel:`Today`, and :guilabel:`Future`. Clicking on each of these buttons shows a
```

File: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+filtered Gantt view of the respective |ECOs|.
+
+.. example::
+   Scheduled activities are shown in the :guilabel:`🕘 (clock)` icon, with `5` |ECOs| pending
```
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst:105]:
```suggestion
   Scheduled activities are shown as a number on the :guilabel:`🕘 (clock)` icon, with `5` |ECOs| pending
```

File: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+   approval :guilabel:`Today`. Currently, there are `0` :guilabel:`Late` or :guilabel:`Future`
+   |ECOs|.
+
+    .. image:: approvals/todo-list.png
+       :align: center
+       :alt: Show scheduled approvals notifications for the user.
+
+Click each pending |ECO|, and a *planned activity* for :guilabel:`ECO Approval` is recorded in the
```
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst:113]:
```suggestion
By clicking a pending |ECO|, a *planned activity* for :guilabel:`ECO Approval` is recorded in the
```

File: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+chatter. Clicking on the :guilabel:`i (Info)` icon to view additional information, including the
```
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst:114]:
```suggestion
chatter. Click on the :guilabel:`i (Info)` icon to view additional information, including the
```

File: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+approval's :guilabel:`Created` date, the approver :guilabel:`Assigned to` it, and the due date.
+
+.. image:: approvals/planned-activity.png
+   :align: center
+   :alt: Show additional details of the planned ECO approval.
+
+Follow-up activities
+~~~~~~~~~~~~~~~~~~~~
+
+When |ECOs| are rejected, tasks need be assigned project members for required modifications before
```
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst:124]:
```suggestion
When |ECOs| are rejected, tasks need to be assigned to project members for required modifications before
```

**Review by hojo-odoo — COMMENTED** [review]
Hey @Felicious,

Great job on this. I left some suggestions, mostly related to phrasing improvements. Also, make sure that |ECO| and |ECOs| appear with the lines on each side every time they're written.

Thanks,
hojo

File: content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+#. :guilabel:`X Engineering Changes` button displays a count of in-progress |ECOs| of this ECO type.
+   Clicking the button opens the Gantt view of the :guilabel:`Engineering Change Orders` page.
+
+#. :guilabel:`My Validations` displays a count of |ECOs| that the approver must accept or reject.
+   Clicking on this button displays |ECOs| pending approval or rejected (marked with the red
+   :guilabel:`Blocked` blocked state).
+
+#. :guilabel:`All Validations` displays a count of |ECOs| a verification stage pending approval from
+   any approvers. Clicking on the button displays |ECOs| pending approval from or rejected by any
+   approver.
```
Comment (Felicious) [inline content/applications/inventory_and_mrp/manufacturing/plm/approvals.rst:65]:
```suggestion
The  :guilabel:`All Validations` button shows the count of |ECOs| awaiting approval or rejected by any approver. Clicking it reveals these pending |ECOs|.```
```

**Comment by Felicious** [comment]
Hi @odoo/crm-doc-review :smile: 

This PLM doc is ready for your review! Just letting you know that there should be 4 build errors, related to missing files/custom anchors for the engineering change orders and version control docs that  should be published before this one. 

Anyways, thank you in advance for giving this doc a review! :grin:

File: content/applications/inventory_and_mrp/plm/management/approvals.rst [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+   .. image:: approvals/approvers.png
+      :align: center
+      :alt: Set an approver that "Is required to approve" ECOs in the "Validated" stage.
+
+Manage approvals
+================
+
+Approvers can easily track their to-do approvals by navigating to the :menuselection:`PLM app` and
+looking at the card for an ECO type, which shows the count of open tasks assigned to them. Here's
+what each button on an ECO project card does:
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/plm/management/approvals.rst:55]:
```suggestion
Approvers can easily track their to-do approvals by navigating to the :menuselection:`PLM app`, and
looking at the card for an ECO type, which shows the count of open tasks assigned to them.

Here's what each button on an ECO project card does:
```

File: content/applications/inventory_and_mrp/plm/management/approvals.rst [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+#. The :guilabel:`# Engineering Changes` button displays a count of in-progress |ECOs| of this ECO
+   type. Clicking the button opens the Gantt view of the :guilabel:`Engineering Change Orders` page.
+
+#. :guilabel:`My Validations` displays a count of |ECOs| that the approver must accept or reject.
+   Clicking on this button displays |ECOs| pending approval or rejected (marked with the red
+   :guilabel:`Blocked` state).
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/plm/management/approvals.rst:62]:
```suggestion
#. :guilabel:`My Validations` displays a count of |ECOs| the approver must accept or reject.
   Clicking on this button displays |ECOs| pending approval or rejected (marked with the red
   :guilabel:`Blocked` state).
```
```suggestion
#. :guilabel:`My Validations` displays a count of |ECOs| that the approver must accept or reject.
   Clicking on this button displays |ECOs| pending approval or rejected (marked with the red
   :guilabel:`Blocked` state).
```

File: content/applications/inventory_and_mrp/plm/management/approvals.rst [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+
+#. The :guilabel:`All Validations` button shows the count of |ECOs| awaiting approval or rejected by
+any approver. Clicking it reveals these pending |ECOs|.
+
+#. :guilabel:`To Apply` displays a count of |ECOs| the user needs to apply changes to. Clicking on
+   the button displays all the |ECOs| to approve and apply changes to in the verification stage.
+   |ECOs| marked with the green :guilabel:`Done` stage have already been approved, and the user just
+   needs to click on the |ECO| to enter the form view and click the :guilabel:`Apply Changes`
+   button.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/plm/management/approvals.rst:71]:
```suggestion
#. :guilabel:`To Apply` displays a count of |ECOs| to which the user needs to apply changes.
   Clicking on the button displays all the |ECOs| to approve, and apply changes to, in the
   verification stage.
   
   |ECOs| marked with the green :guilabel:`Done` stage have already been approved, and the user just
   needs to click on the |ECO| to enter the form view, and click the :guilabel:`Apply Changes`
   button.
```

File: content/applications/inventory_and_mrp/plm/management/approvals.rst [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+
+Approve ECOs
+------------
+
+Navigate to an |ECO| in a verification stage while logged in as the assigned approver to see the
+:guilabel:`Approve`, :guilabel:`Reject`, and :guilabel:`Apply Changes` buttons. To approve the |ECO|
+and apply the changes onto the production :abbr:`BoM (Bill of Materials)`, click
+:guilabel:`Approve`, and then :guilabel:`Apply Changes`. Note that the :guilabel:`Apply Changes`
+button will not work unless the :guilabel:`Approve` button was clicked first. Additionally, the
+chatter will log the history of the buttons clicked.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/plm/management/approvals.rst:85]:
```suggestion
Navigate to an |ECO| in a verification stage, while logged in as the assigned approver, to see the
:guilabel:`Approve`, :guilabel:`Reject`, and :guilabel:`Apply Changes` buttons.

To approve the |ECO|, and apply the changes onto the production :abbr:`BoM (Bill of Materials)`,
click :guilabel:`Approve`, and then :guilabel:`Apply Changes`.

Note that the :guilabel:`Apply Changes` button will **not** work unless the :guilabel:`Approve`
button was clicked first. Additionally, the chatter logs the history of the clicked buttons.
```

File: content/applications/inventory_and_mrp/plm/management/approvals.rst [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+   :align: center
+   :alt: Show additional details of the planned ECO approval.
+
+Follow-up activities
+~~~~~~~~~~~~~~~~~~~~
+
+When |ECOs| are rejected, tasks need to be assigned to project members for required modifications
+before |ECO| approval. To create tasks with deadlines, navigate to the rejected |ECO| form and go to
+the chatter. Select the :guilabel:`Mark Done` button in the :guilabel:`Planned Activities` section
+of the chatter to close the activity and open a pop-up window for creating tasks.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/plm/management/approvals.rst:127]:
```suggestion
before |ECO| approval. To create tasks with deadlines, navigate to the rejected |ECO| form, and go
to the chatter.

Select the :guilabel:`Mark Done` button in the :guilabel:`Planned Activities` section of the chatter
to close the activity, and open a pop-up window for creating tasks.
```

File: content/applications/inventory_and_mrp/plm/management/approvals.rst [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+         close the planned activity.
+
+In the :guilabel:`Mark Done` window, click :guilabel:`Done & Schedule Next` to open a new
+:guilabel:`Schedule an Activity` window. Next, set the :guilabel:`Assigned to` team member and the
+:guilabel:`Due Date` for completing the changes. Provide task details in the :guilabel:`Summary`
+field and the text box. Click the :guilabel:`Schedule` button to close the window.
+
+After closing the window, on the |ECO| form, move the |ECO| back one stage. Doing so ensures that
+when the team member completes the changes and returns the |ECO| to the verification stage, a new
+:guilabel:`ECO Approval` task is created for the approver.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/plm/management/approvals.rst:141]:
```suggestion
when the team member completes the changes, and returns the |ECO| to the verification stage, a new
:guilabel:`ECO Approval` task is created for the approver.
```

**Review by ksc-odoo — COMMENTED** [review]
Great work @Felicious -- just finished my peer review of this PR and only had a few minor suggestions that'll need your attention. But, once you give those a look, I think this will be ready for the next stage. Once again, Awesome Job with this one! 👍

**Review by StraubCreative — APPROVED** [review]
Another great doc @Felicious 🤙 

@robodoo r+

**Comment by StraubCreative** [comment]
@robodoo r+

**Pull request closed without merging**